src/researchhub_document/hot_score_mixin2.py

```diff
@@ -0,0 +1,151 @@
+import datetime
+import math
+from researchhub_document.related_models.constants.document_type import PAPER
+import numpy as np
+from django.db.models import (
+    Q,
+    Count
+)
+from discussion.models import Vote
+
+
+class HotScoreMixin:
+
+    def _c(self, num):
+        if num >= 0:
+            return 1
+        else:
+            return -1
+
+    def _count_doc_comment_votes(self, doc):
+        vote_total = 0
+        try:
+            for t in doc.threads.filter(is_removed=False).iterator():
```

Comment:
do we have unified_doc linked to comments / threads / replies? if so then it would be easier to just filter by the unified doc or paper

Reply:
@lightninglu10 I don't think so but I shall check!